test/bats/azure.bats

```diff
@@ -125,13 +126,70 @@ setup() {
   [[ "${result//$'\r'}" -eq "${SECRET_VALUE}" ]]
 
   result=$(kubectl get secret foosecret -o json | jq '.metadata.ownerReferences | length')
-  [[ "$result" -eq "2" ]]
+  [[ "$result" == "2" ]]
 }
 
 @test "Sync with K8s secrets - delete deployment, check secret deleted" {
   run kubectl delete -f $BATS_TESTS_DIR/nginx-deployment-synck8s-azure.yaml
   assert_success
+
+  run kubectl delete -f $BATS_TESTS_DIR/azure_synck8s_v1alpha1_secretproviderclass.yaml
+  assert_success
+
   sleep 20
   result=$(kubectl get secret | grep foosecret | wc -l)
   [[ "$result" -eq "0" ]]
 }
+
+@test "Test Namespaced scope SecretProviderClass - create deployment" {
+  run kubectl create ns test-ns
+  assert_success
+
+  run kubectl create secret generic secrets-store-creds --from-literal clientid=${AZURE_CLIENT_ID} --from-literal clientsecret=${AZURE_CLIENT_SECRET} -n test-ns
+  assert_success
+
+  envsubst < $BATS_TESTS_DIR/azure_v1alpha1_secretproviderclass_ns.yaml | kubectl apply -f -
+
+  cmd="kubectl wait --for condition=established --timeout=60s crd/secretproviderclasses.secrets-store.csi.x-k8s.io"
+  wait_for_process $WAIT_TIME $SLEEP_TIME "$cmd"
+
+  cmd="kubectl get secretproviderclasses.secrets-store.csi.x-k8s.io/azure-sync -o yaml | grep azure"
+  wait_for_process $WAIT_TIME $SLEEP_TIME "$cmd"
+
+  cmd="kubectl get secretproviderclasses.secrets-store.csi.x-k8s.io/azure-sync -n test-ns -o yaml | grep azure"
```

Comment:
is there also a negative test for referring to a spc in a different ns?

Reply:
Added a negative test case that deploy pods in `negative-test-ns`. The mount will fail with 

```
  Warning  FailedMount  3s (x8 over 67s)  kubelet, aks-nodepool1-36438484-vmss000001  MountVolume.SetUp failed for volume "secrets-store-inline" : kubernetes.io/csi: mounter.SetupAt failed: rpc error: code = Unknown desc = failed to get secretproviderclass negative-test-ns/azure-sync, error: secretproviderclasses.secrets-store.csi.x-k8s.io "azure-sync" not found
```